angular/src/lib/input-group/input-group.component.ts

```diff
@@ -0,0 +1,278 @@
+      <md-input-message *ngFor="let message of messages">
+        {{message}}
+      </md-input-message>
+      <ng-content select="md-input-message"></ng-content>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [InputService],
+})
+
+export class InputGroupComponent implements AfterContentInit, AfterViewInit, OnChanges, OnDestroy {
+  private _explicitFormFieldControl: InputGroupService<any>;
+  private _destroyed = new Subject<void>();
+  protected _clear: boolean = false;
+  protected _status: string = '';
+  protected _isError: boolean = false;
+  public messages: Array<[]>;
+
+   // Unique id for the hint label.
+   _inputHelpTextId: string = `md-input__help-text-${nextUniqueId++}`;
+
+  /** @prop Optional css class string | ''  */
+  @Input() public class: string = '';
+  /** @prop Optional button to clear input text | false */
+  @Input()
+  get clear(): boolean { return this._clear; }
+  set clear(value: boolean) {
+    this._clear = value || false;
+  }
+  /** @prop Optional clear aria label string | 'clear'  */
+  @Input() public clearAriaLabel: string = 'clear';
+  /** @prop Overall input container size | '' */
+  @Input() public containerSize: string = '';
+  /** @prop Help Text to appear under the input | '' */
+  @Input() public inputHelpText: string = '';
+  /** @prop Input wrapper size | '' */
+  @Input() public inputSize: string = '';
+  /** @prop Sets the filled class of the Input Container | false */
+  @Input() public isFilled: boolean = false;
+  /** @prop Input label text | '' */
+  @Input() public label: string = '';
+  /** @prop Array of objects with error type and error message */
+  @Input() public messageArr: any[];
+  /** @prop Set the level of nested Input components | 0 */
+  @Input() public nestedLevel: number = 0;
+  /** @prop Secondary Input label | ''  */
+  @Input() public secondaryLabel: string = '';
+  /** @prop Optional status to override error status | '' */
+  @Input()
+  get status(): string { return this._status; }
+  set status(value: string) {
+    this._status = value || '';
+  }
+
+  // Add static argument once Angular version is updated
+  @ContentChild(InputGroupService) _controlNonStatic: InputGroupService<any>;
+  @ContentChild(InputGroupService) _controlStatic: InputGroupService<any>;
+  @ContentChildren(InputSectionComponent) _iconChildren: QueryList<InputSectionComponent>;
+  @ContentChildren(InputMessageComponent) _contentMessageChildren: QueryList<InputMessageComponent>;
+  @ViewChildren(InputMessageComponent) _viewMessageChildren: QueryList<InputMessageComponent>;
+
+  get _control() {
+    // TODO(crisbeto): we need this hacky workaround in order to support both Ivy
+    // and ViewEngine. We should clean this up once Ivy is the default renderer.
+    return this._explicitFormFieldControl || this._controlNonStatic || this._controlStatic;
```

Comment:
Um .. maybe we shouldn't copy this exactly from material that we still have their TODO's

Reply:
haha agreed - I left it there for the intent of the message since that would apply to us as well but I can remove the name.